modules/javafx.graphics/src/main/java/javafx/scene/layout/BorderPane.java

```diff
@@ -389,7 +391,8 @@ public final ObjectProperty<Node> rightProperty() {
             double topPrefHeight = getAreaHeight(getTop(), -1, false);
             double bottomPrefHeight = getAreaHeight(getBottom(), -1, false);
 
-            double middleAreaHeight = Math.max(0, height - topPrefHeight - bottomPrefHeight);
+            double middleAreaHeight = Math.max(0,
+                    height - insets.getTop() - insets.getBottom() - topPrefHeight - bottomPrefHeight);
```

Comment:
should these be snapped?  snappedBottomInset() etc.?

Reply:
You might be right. 

In this case, `topPrefHeight` comes from `getAreaHeight()`, that calls `computeChildPrefAreaHeight()` that ultimately uses `snapSpaceY()`. 

However, this would also mean that the returned value should use snapped insets as well?
```
return insets.getLeft() +
                Math.max(leftPrefWidth + centerMinWidth + rightPrefWidth, Math.max(topMinWidth,bottomMinWidth)) +
                insets.getRight();
```
and also for the layout call?

Reply:
no, I did not propose to change it everywhere.
In this PR, in the code we change, we probably should [start to] use snapped coordinates.  wouldn't you agree?

Reply:
Hard to say. Unless it's somehow related to the changes being made, I can see a good argument for addressing the snapping in a follow-up issue, so as not to introduce what is essentially an unrelated fix.